include/SFML/Window/Event.inl

```diff
@@ -0,0 +1,64 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2023 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+Event::Event(const T& t)
+{
+    static_assert(isEventType<T>, "T must be a subtype of sf::Event");
+    if constexpr (isEventType<T>)
+        m_data = t;
+}
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+bool Event::is() const
+{
+    static_assert(isEventType<T>, "T must be a subtype of sf::Event");
+    if constexpr (isEventType<T>)
+        return std::holds_alternative<T>(m_data);
+}
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+const T& Event::get() const
+{
+    static_assert(isEventType<T>, "T must be a subtype of sf::Event");
+    assert(is<T>() && "Event::get() Cannot access an inactive event type");
+    if constexpr (isEventType<T>)
+        return std::get<T>(m_data);
```

Comment:
Consider: 

```suggestion
        return *std::get_if<T>(&m_data);
```

This likely generates better code and results in UB in case `!is<T>()`, which exposes even less details about variant and prevents users from thinking that "catching" the exception might be a way of handling the error.

Reply:
I'm cool with that. That way we can remove a reference to `std::bad_variant_access` from the header. We've set a precedent that the library won't go to lengths to guard against UB in release builds and since we're still protecting against UB in debug builds via the `assert` that's acceptable to me.